clusterloader2/pkg/measurement/common/generic_query_measurement_test.go

```diff
@@ -150,6 +169,158 @@ func TestGather(t *testing.T) {
+			},
+			samples: map[string][]*model.Sample{
+				"query-perc99[1m]": {
+					{
+						Metric: model.Metric{
+							model.LabelName("d1"): model.LabelValue("d1-val1"),
+							model.LabelName("d2"): model.LabelValue("d2-val1"),
+							model.LabelName("d3"): model.LabelValue("d3-val1"), // Ignored
+						},
+						Value: model.SampleValue(1),
+					},
+					{
+						Metric: model.Metric{
+							model.LabelName("d1"): model.LabelValue("d1-val1"),
+							model.LabelName("d2"): model.LabelValue("d2-val2"),
+						},
+						Value: model.SampleValue(2),
+					},
+				},
+				"query-perc90[1m]": {
+					{
+						Metric: model.Metric{
+							model.LabelName("d1"): model.LabelValue("d1-val1"),
+							model.LabelName("d2"): model.LabelValue("d2-val1"),
+						},
+						Value: model.SampleValue(3),
+					},
+					{
+						Metric: model.Metric{
+							model.LabelName("d1"): model.LabelValue("d1-val1"),
+							model.LabelName("d2"): model.LabelValue("d2-val2"),
+						},
+						Value: model.SampleValue(4),
+					},
+					{
+						Metric: model.Metric{
+							model.LabelName("d1"): model.LabelValue("d1-val1"),
+							// d2 not set
+						},
+						Value: model.SampleValue(5),
+					},
+				},
+			},
+			wantDataItems: []measurementutil.DataItem{
+				{
+					Labels: map[string]string{
+						"MetricName": "dimensions",
+						"d1":         "d1-val1",
+						"d2":         "d2-val1",
+					},
+					Unit: "ms",
+					Data: map[string]float64{
+						"perc99": 1.0,
+						"perc90": 3.0,
+					},
+				},
+				{
+					Labels: map[string]string{
+						"MetricName": "dimensions",
+						"d1":         "d1-val1",
+						"d2":         "d2-val2",
+					},
+					Unit: "ms",
+					Data: map[string]float64{
+						"perc99": 2.0,
+						"perc90": 4.0,
+					},
+				},
+				{
+					Labels: map[string]string{
+						"MetricName": "dimensions",
+						"d1":         "d1-val1",
+						"d2":         "",
+					},
+					Unit: "ms",
+					Data: map[string]float64{
+						// perc99 doesn't return this combination.
+						"perc90": 5.0,
+					},
+				},
+			},
+		},
+		{
+			desc: "multiple values for single dimension",
+			params: map[string]interface{}{
+				"metricName":    "dimensions",
+				"metricVersion": "v1",
+				"unit":          "ms",
+				"dimensions": []interface{}{
+					"d1",
+					"d2",
+				},
+				"queries": []map[string]interface{}{
+					{
+						"name":  "perc99",
+						"query": "query-perc99[%v]",
+					},
+				},
+			},
+			samples: map[string][]*model.Sample{
+				"query-perc99[1m]": {
+					{
+						Metric: model.Metric{
+							model.LabelName("d1"): model.LabelValue("d1-val1"),
+							model.LabelName("d2"): model.LabelValue("d2-val1"),
+						},
+						Value: model.SampleValue(1),
+					},
+					{
+						Metric: model.Metric{
+							model.LabelName("d1"): model.LabelValue("d1-val1"),
+							model.LabelName("d2"): model.LabelValue("d2-val1"),
+						},
+						Value: model.SampleValue(2),
+					},
+				},
+			},
+			wantErr: "too many samples",
```

Comment:
Why we do not need full error message here?

Reply:
Bug in test code. I actually left the substring behaviour to avoid change detector tests.